Source/WebCore/bindings/js/SerializedScriptValue.cpp

```diff
@@ -1725,7 +1732,15 @@ class CloneSerializer : CloneBase {
 
     void write(const File& file)
     {
-        m_blobHandles.append(file.handle().isolatedCopy());
+        SecurityOriginData topOrigin;
+        auto partitionBlobRegistry { PartitionBlobRegistry::No };
+        if (auto* context = executionContext(m_lexicalGlobalObject)) {
+            topOrigin = context->topOrigin().data();
+            partitionBlobRegistry = context->settingsValues().blobRegistryTopOriginPartitioningEnabled ? PartitionBlobRegistry::Yes : PartitionBlobRegistry::No;
```

Comment:
I think for this particular use case, given that the blob registry is global and in the network process, it would be nice to have a global setting in the network process, rather that pass the setting for each blob registry function call in the WebProcess.

You can still use an experimental feature flag but we'd basically add logic in the UIProcess to deduce the global network process setting for the pages settings. Like, if any page has partitioning enabled, then we should enable partitioning in the network process. Then the WebProcess wouldn't have to worry about the setting at all. I think this should simplify things nicely.

I also discussed it offline with Brady and he seemed onboard with this idea.